### PR TITLE
Upgrade packages and move to ESLint

### PR DESCRIPTION
**Pull request by benjay10**
Remove old deprecated TSLint and move to ESLint.

As a newcomer to TypeScript, I'm not familiar with the tools. Trying to run the code from this repository led to package dependency problems, probably caused by newer version of TypeScript conflicting the maximum supported versions of TypeScript compatible with the TSLint system. With this PR I tried to move to ESLint instead with the same, mostly default configuration, update some other packages, and solve some minor linting issues at once.

### CODE REVIEW

**Comment by rubensworks** [comment]
Thanks for this @benjay10! A migration to eslint has indeed been needed for a while.

However, to have some consistency across projects, I always use the same eslint config within my projects.
Would it be possible to copy the eslint setup as used here: https://github.com/rubensworks/rdf-data-factory.js ?

**Comment by benjay10** [comment]
I think this is now more in line with the rest of your projects.

By introducing the new ESLint configuration, there were hundreds of linting problems, of course. I hope nothing has broken. Probably not as all the tests were still correct.
There is also still some problem with package dependency version. Your `@rubensworks/eslint-config` relies on older `@typescript-eslint/eslint-plugin` (and other) packages, so I had to downgrade them here.

File: package.json [hunk 1]
```diff
@@ -59,16 +67,27 @@
...
     "test-watch": "jest ${1} --watch",
     "coveralls": "jest --coverage && cat ./coverage/lcov.info | coveralls",
-    "lint": "tslint index.ts lib/**/*.ts test/**/*.ts --exclude '**/*.d.ts'",
+    "lint": "eslint . --ext .ts --cache",
     "build": "tsc",
     "validate": "npm ls",
     "prepare": "npm run build",
     "version": "manual-git-changelog onversion"
+  },
+  "resolutions": {
```
Comment (rubensworks) [inline package.json]: This can be removed.

**Review by rubensworks — CHANGES_REQUESTED** [review]
Looks perfect!
Just one minor thing, then this should be good to go 🙂

File: package.json (hunk 1 above)
Comment (benjay10) [reply package.json]: Done

**Comment by rubensworks** [comment]
Thanks @benjay10!

**Pull request merged**